docs/DeveloperGuide.md

```diff
@@ -1,38 +1,442 @@
-|v1.0|new user|see usage instructions|refer to them when I forget how to use the application|
-|v2.0|user|find a to-do item by name|locate a to-do without having to go through the entire list|
 
-## Non-Functional Requirements
+## CLI Applications
 
-{Give non-functional requirements}
+The idea for this product is for users who are familiar with CLI environments. Users are able to see only the necessary information given in text format. Things can be done faster with just a single line of input command compared to navigating through a GUI environment.
 
-## Glossary
 
-* *glossary item* - Definition
+---
 
-## Instructions for manual testing
 
-{Give instructions on how to do a manual product testing e.g., how to load sample data to be used for testing}
+# Setting up, getting started
+
+
+## Setting up the project in your computer
+
+First, fork this repo, and clone the fork into your computer.
+
+If you plan to use Intellij IDEA (highly recommended):
+
+
+
+1. **Configure the JDK:** Follow the guide [[se-edu/guides] IDEA: Configuring the JDK](https://se-education.org/guides/tutorials/intellijJdk.html) to ensure Intellij is configured to use JDK 11.
+2. **Import the project as a Gradle project:** Follow the guide [[se-edu/guides] IDEA: Importing a Gradle project](https://se-education.org/guides/tutorials/intellijImportGradleProject.html) to import the project into IDEA. \
+
+<img src="./DGDiagramsCreator/DGUsedDiagrams/ExclamationMark.png" width="4%" />
+Note: Importing a Gradle project is slightly different from importing a normal Java project.
+3. **Verify the setup:**
+    1. Run the _seedu.address.Main_ and try a few commands.
+    2. [Run the tests](https://se-education.org/addressbook-level3/Testing.html) to ensure they all pass.
+
+
+## Before writing code
+
+
+
+1. **Configure the coding style** \
+   If using IDEA, follow the guide [[se-edu/guides] IDEA: Configuring the code style](https://se-education.org/guides/tutorials/intellijCodeStyle.html) to set up IDEA’s coding style to match ours. 
+
+<img src="./DGDiagramsCreator/DGUsedDiagrams/ExclamationMark.png" width="4%" />
+Tip: Optionally, you can follow the guide [[se-edu/guides] Using Checkstyle](https://se-education.org/guides/tutorials/checkstyle.html) to find how to use the CheckStyle within IDEA e.g., to report problems as you write code.  
+
+2. **Set up CI** This project comes with a GitHub Actions config files (in .github/workflows folder). When GitHub detects those files, it will run the CI for your project automatically at each push to the master branch or to any PR. No set up required.  
+3. **Learn the design** When you are ready to start coding, we recommend that you get some sense of the overall design by reading about [AddressBook’s architecture](https://se-education.org/addressbook-level3/DeveloperGuide.html#architecture).  
+4. **Do the tutorials These tutorials will help you get acquainted with the codebase.**  
```

Comment:
Are these numberings updated in the page?
![image](https://user-images.githubusercontent.com/86601208/228775319-a71168d5-16f0-4d80-9604-88126ee07848.png)